Updating dev doc about labels and milestones (#6661)
DataDog/datadog-agent: docs/dev/contributing.md

    changes, cross-referencing any related bugs/PRs.
  * use [Reno](#reno) to create a releasenote.
  * open your PR against the `master` branch.

>   * for internal PR from Datadog (we will set them when reviewing for community PRs):

Reviewer: ```suggestion
  * for PRs from contributors with write access to the repository (for community PRs, will be done by Datadog employees):
```

>     + set the `team/agent-core` label

Reviewer: ```suggestion
    + set the relevant `team/` label
```

>     + add a milestone to your PR (use the highest available, ex: `6.8.0`)

Reviewer: ```suggestion
    + add a milestone to your PR (by default, use the highest milestone version available, ex: `6.8.0`)
```

      Agent 5's behavior.
  ```

## PR labels

A few extra labels can be set on your PR:
- `community/help-wanted`: for community PRs where help is needed to finish it.
- `community`: for community PRs.

> - `changelog/no-changelog`: for PRs who don't require a reno releasenote

Reviewer: ```suggestion
- `changelog/no-changelog`: for PRs that don't require a reno releasenote
```

      Agent 5's behavior.
  ```

## PR labels

For internal PRs (from people in the Datadog organisation), you have few extra
labels that can be use:
- `community/help-wanted`: for community PRs where help is needed to finish it.
- `community`: for community PRs.
- `changelog/no-changelog`: for PRs who don't require a reno releasenote
  (useful for PRs only changing documentation or tests).
- `qa/skip-qa`: this will skip creating a QA card for the PR during the release
  process (example: for a documentation only PRs).
- `major_change`: to flag the PR as a major change impacting many/all teams
  working on the agent and will require deeper QA (example: when we change the

>   Python version ship in the agent).

Reviewer: ```suggestion
  Python version shipped in the agent).
```
